src/InstrumentationEngine/InstructionGraph.cpp

```diff
@@ -17,6 +17,15 @@ MicrosoftInstrumentationEngine::CInstructionGraph::~CInstructionGraph()
         CCriticalSectionHolder lock(&m_cs);
 
         CComPtr<CInstruction> pInstr = (CInstruction*)m_pFirstInstruction;
+
+        m_pFirstInstruction.Release();
+        m_pOrigFirstInstruction.Release();
+        
+        if(m_pUninstrumentedFirstInstruction != nullptr)
+        {
+            m_pUninstrumentedFirstInstruction.Release();
+        }
+
         while (pInstr != nullptr)
```

Comment:
You can verify if you have any leaking instruction objects by using Debug mode bits of clrie and having a debugger attached during shutdown. See [refcount.cpp](https://github.com/microsoft/CLRInstrumentationEngine/blob/develop/src/InstrumentationEngine.Lib/refcount.cpp)